experiments/experiment_prototype.py

```
@@ -1,385 +1,1111 @@
+                else:  # no break, so no changes to the clrfrqrange
+                    still_checking = False
+
+        elif exp_slice['rxonly']:  # RX only mode.
+            # In this mode, rxfreq is required.
+            try:
+                assert isinstance(exp_slice['rxfreq'], int) or isinstance(exp_slice['rxfreq'],
+                                                                          float)
+                assert (exp_slice['rxfreq'] * 1000) < self.options.max_freq
+                assert (exp_slice['rxfreq'] * 1000) > self.options.min_freq
+            except AssertionError:
+                errmsg = """rxfreq must be a number (kHz) between min and max frequencies for the 
+                            radar"""
+                raise ExperimentException(errmsg)
+
+        else:  # TX-specific mode , without a clear frequency search.
+            # In this mode, txfreq is required along with the other requirements.
+            try:
+                assert isinstance(exp_slice['txfreq'], int) or isinstance(exp_slice['txfreq'],
+                                                                          float)
+                assert (exp_slice['txfreq'] * 1000) < self.options.max_freq
+                assert (exp_slice['txfreq'] * 1000) > self.options.min_freq
+            except AssertionError:
+                errmsg = """txfreq must be a number (kHz) between min and max frequencies for the 
+                                            radar"""
+                raise ExperimentException(errmsg)
+            for freq_range in self.options.restricted_ranges:
+                try:
+                    assert exp_slice['txfreq'] not in range(freq_range[0], freq_range[1])
+                except AssertionError:
+                    errmsg = """txfreq is within a restricted frequency range"""
+                    raise ExperimentException(errmsg)
+                    # TODO or could just log this and set to default frequency, or the closest
```

We like the idea of failing early if txfreq is restricted, thinking about the situation with epop where we didn't notice the default freq being used, then the experiment was basically useless.